scripts/run_mandiff.sh

```diff
@@ -29,110 +61,58 @@ cd "${ROOT}"
 
 export GO111MODULE=on
 # build the istio operator binary
-go build -o $GOPATH/bin/mesh ./cmd/mesh.go
+go build -o "${GOPATH}/bin/mesh" ./cmd/mesh.go
 
 # download the helm binary
 ${ROOT}/bin/init_helm.sh
 
-function helm_render_template() {
-    local namespace=${1}
-    shift
-    local relase=${1}
-    shift
-    local chart=${1}
-    shift
-    local cfg=${1}
-    shift
-
-    # render the chart with helm template
-    helm template --namespace $namespace --name $relase $chart $cfg $*
-}
-
 # render all the templates with helm template.
 function helm_manifest() {
-    local namespace=${1}
-    shift
-    local relase=${1}
-    shift
-    local chart=${1}
-    shift
-    local profile=${1}
-    shift
+    local namespace="${1}"
+    local release="${2}"
+    local chart="${3}"
+    local profile="${4}"
 
     # the global settings are the default for the chart
-    local cfg="-f ${chart}/global.yaml"
-
     # the specified profile will override the gloal settings
-    if [ -f "${ROOT}/tests/profiles/helm/values-istio-${profile}.yaml" ]; then
-        cfg="${cfg} -f ${ROOT}/tests/profiles/helm/values-istio-${profile}.yaml"
-    else
-        echo "Please verify the values file for ${profile} path exists."
-        exit 1
-    fi
+    local cfg="-f ${chart}/global.yaml -f ${ROOT}/tests/profiles/helm/values-istio-${profile}.yaml"
 
     # create parent directory for the manifests rendered by helm template
-    mkdir -p ${OUT}/helm-template/istio-${namespace}-${relase}-${profile}
+    local out_dir="${OUT}/helm-template/istio-${profile}"
+    mkdir -p "${out_dir}"
 
-    local charts=${profile_charts_map[${profile}]}
+    local charts="${PROFILE_CHARTS_MAP[${profile}]}"
     for c in $(echo $charts | tr " " "\n")
     do
-        # create the parent directory if it doesn't exist.
-        mkdir -p $(dirname ${OUT}/helm-template/istio-${namespace}-${relase}-${profile}/${c}.yaml)
-        helm_render_template ${namespace} ${relase} ${chart}/${c} ${cfg} $* > ${OUT}/helm-template/istio-${namespace}-${relase}-${profile}/${c}.yaml
+       echo "Rendering ${c}"
+       mkdir -p "${out_dir}/${c}"
```

Comment:
This will create some empty directory. Can we replace with `mkdir -p $(dirname "${out_dir}/${c}.yaml")`?

Reply:
You mean leave out the subdirectory, since the filename contains the component name? We could do that but I was trying to be consistent with the layout generated with mesh manifest generate for easier diffing in meld.

Reply:
That makes sense.